### DISCUSSION TITLE
How to use Flux and Mono as the return type of methods annotated by @DgsQuery and @DgsMutation

### DISCUSSION THREAD

**Discussion by shiyouping** [Netflix/dgs-framework#374] [Q&A] [answered]
I tried to use Flux and Mono from projectreactor as the return type of methods annotated by @DgsQuery in the graphql-dgs-webflux-starter project, but it doesn't work. The query always returns null. I already compared [webflux example](https://github.com/Netflix/dgs-examples-webflux) and [mvc example](https://github.com/Netflix/dgs-examples-java), and it shows that methods annotated by @DgsQuery are almost the same. Looks like it is not necessary to use webflux. Could you please let me know if it's possible to use Flux and Mono with @DgsQuery and @DgsMutation? If possible, how? Thanks in advance.

**Comment by ghost** [comment]
I have exactly the same question. The webflux example is almost a copy paste of the mvc but with another library. It doesn't return any Mono or Flux or it doesn't contain any test with the new Executor. I also couldn't find any way of returning a Flux or Mono in my project. I still need to return CompletableFutures...
Any help would be really appreciated.

**Comment by paulbakker** [comment]
This was out of scope for the initial implementation, but it's an obvious need.
There's an issue about this as well: https://github.com/Netflix/dgs-framework/issues/376 and I just commented there with a possible PR. Please reply to the issue with further thoughts!

**Comment by shiyouping** [comment, marked as answer]
Already tested with `com.netflix.graphql.dgs:graphql-dgs-webflux-starter:4.3.1`, Mono and Flux are supported now.

**Comment by gianielsevier** [comment]
Will the BatchLoader support Mono or Flux instead of CompletionStage in the future?

**Reply by srinivasankavitha** [reply]
BatchLoader APIs come from graphql-java's dataloader library. We don't plan on making any updates in the future unless there are changes there. In addition, we now have spring-graphql integration, so for webflux, that would be preferable. We don't plan on  making enhancements to our webflux implementation since it is not internally used.
